### PR TITLE
Support 19138 Create OpenAPI 3 spec for Agent module

### PR DESCRIPTION
**Pull request by xuemeigan**

### CODE REVIEW

File: schema/agent.yaml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    post:
+      summary: adds an agent
+      operationId: addAgent
+      description: Adds an agent
+      responses:
+        '201':
+          description: agent created
+        '400':
+          description: 'invalid input, object invalid'
+        '409':
```
Comment (Jkeyuk) [inline schema/agent.yaml]: 422

File: schema/agent.yaml [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            type: string
+            format: uuid            
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Agent'
+        "400":
```
Comment (Jkeyuk) [inline schema/agent.yaml]: 500 "not a valid uuid"

File: schema/agent.yaml [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+          in: path
+          description: agent id to delete
+          required: true
+          schema:
+            type: string
+            format: uuid            
+      responses:
+        '200':
+          description: agent marked as deleted
+        '400':
```
Comment (Jkeyuk) [inline schema/agent.yaml]: 500 "not a valid uuid"

**Review by Jkeyuk — COMMENTED** [review]
Added a few comments describing a few differences between defined/actual response codes

File: schema/agent.yaml (hunk 1 above)
Comment (xuemeigan) [reply schema/agent.yaml]: Unprocessible entity

**Pull request merged**